app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -806,19 +807,25 @@ public void startLoading(final boolean forceLoad) {
             currentWorker.dispose();
         }
 
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(activity);
+
         currentWorker = ExtractorHelper.getStreamInfo(serviceId, url, forceLoad)
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe((@NonNull StreamInfo result) -> {
+                .subscribe((@NonNull final StreamInfo result) -> {
                     isLoading.set(false);
-                    currentInfo = result;
-                    handleResult(result);
-                    showContent();
-                }, (@NonNull Throwable throwable) -> {
+                    if (result.getAgeLimit() != NO_AGE_LIMIT && !prefs.getBoolean(
+                            getString(R.string.show_age_restricted_content), false)) {
+                        showError(getString(R.string.restricted_video), false);
```

Comment:
Add a function at line 1243, this should fix the tab hiding issues. Comments are shown until the video is loaded, and are then hidden, but there is not much that can be done about this since when comments are loaded we do not yet know if the video is age restricted or not
```java
    private void hideAgeRestrictedContent() {
        showError(getString(R.string.restricted_video), false);

        if (relatedStreamsLayout != null) { // tablet
            relatedStreamsLayout.setVisibility(View.INVISIBLE);
        }

        viewPager.setVisibility(View.GONE);
        tabLayout.setVisibility(View.GONE);
    }
```
```suggestion
                        hideAgeRestrictedContent();
```

Reply:
hum, not sure whether we should hide comments.
Nor PeerTube neither YouTube do that.